### PR TITLE
New rules for PHPUnit test methods, and Variable type hints

### PR DESCRIPTION
**Pull request by CMalvika**

### CODE REVIEW

File: src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        // Visibility
+        $visibilityPtr = $file->findPrevious([T_FUNCTION, T_WHITESPACE, T_STATIC], $namePtr - 1, null, true);
+
+        // Skip invalid statement.
+        if (!isset($tokens[$namePtr + 1]['parenthesis_opener'])) {
+            return;
+        }
+
+        // Skip test methods
+        if (preg_match('/^test/D', $methodName)) {
```
Comment (lstrojny) [inline src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php]: You can use strpos here

File: src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            return;
+        }
+
+        if ($tokens[$visibilityPtr]['code'] === T_PUBLIC) {
+            $file->addError(
+                'All public methods in a PHPUnit test must either start with test* or be data providers. '
+                . 'This is for to make sure, we are not accidentally skipping a test because for example a typo '
+                . '(tsetSomething()).',
+                $stackPtr,
+                'verbLimit'
```
Comment (lstrojny) [inline src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php]: What’s a verb limit?

File: src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                        $dataProvider[$j] =
+                            trim($tokens[$file->findNext(T_DOC_COMMENT_WHITESPACE, $j + 1, null, true)]['content']);
+                    }
+                }
+            }
+        }
+
+        // Class name
+        $classPtr = $file->findPrevious(T_CLASS, $stackPtr);
+        $className = $tokens[$file->findNext(T_WHITESPACE, $classPtr + 1, null, true)]['content'];
```
Comment (lstrojny) [inline src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php]: You can use $file->getIdentifierName($classPtr) here

File: src/InterNations/Sniffs/Syntax/VarTypeHintsSniff.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                return;
+            }
+
+            $varDoc = preg_split(
+                '/[\s]+/',
+                trim($tokens[$file->findNext(T_DOC_COMMENT_WHITESPACE, $j + 1, null, true)]['content'])
+            );
+
+            if (preg_match('/^ArrayCollection/D', $varDoc[0])) {
+                $str = 'Found type "%1$s" for a variable "%2$s", ';
```
Comment (lstrojny) [inline src/InterNations/Sniffs/Syntax/VarTypeHintsSniff.php]: "for property" instead of "for a variable" would be better

Comment (CMalvika) [reply src/InterNations/Sniffs/Syntax/VarTypeHintsSniff.php]: (y)

File: src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php (hunk 2 above)
Comment (CMalvika) [reply src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php]: It's wrong! Copied from somewhere, i will change.

File: src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php (hunk 3 above)
Comment (CMalvika) [reply src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php]: getIdentifierName is not defined.

Comment (lstrojny) [reply src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php]: Sorry, I meant getDeclarationName()

Comment (CMalvika) [reply src/InterNations/Sniffs/Architecture/TestMethodsConventionSniff.php]: yes, thank you. later i will try to modify other sniffs as well.

**Comment by lstrojny** [comment]
@CMalvika can you check the build?

**Pull request merged**